src/compiler/checker.ts

```diff
@@ -2716,6 +2716,11 @@ namespace ts {
                         context.encounteredError = true;
                         return undefined;
                     }
+                    else if (context.flags & NodeBuilderFlags.WriteClassExpressionAsTypeLiteral &&

```

Comment:
Where does this ever get passed in?

Reply:
Ahhhhh..... Nowhere (internally), apparently. Until the swap to the node builder in the other PR, we only use the symbol display builder to print classes as object literals.

Reply:
So this should not be in this change? it should be in the symbolDisplayBuilder change instead...